miner/scroll_worker.go

```diff
@@ -0,0 +1,943 @@
+		if result := w.currentPipeline.TryPushTxns(txs, w.onTxFailingInPipeline); result != nil {
+			w.handleCCCResult(result)
+			return
+		}
+	}
+
+	if len(localTxs) > 0 {
+		txs := types.NewTransactionsByPriceAndNonce(signer, localTxs, header.BaseFee)
+		if result := w.currentPipeline.TryPushTxns(txs, w.onTxFailingInPipeline); result != nil {
+			w.handleCCCResult(result)
+			return
+		}
+	}
+	if len(remoteTxs) > 0 {
+		txs := types.NewTransactionsByPriceAndNonce(signer, remoteTxs, header.BaseFee)
+		if result := w.currentPipeline.TryPushTxns(txs, w.onTxFailingInPipeline); result != nil {
+			w.handleCCCResult(result)
+			return
+		}
+	}
+}
+
+func (w *worker) handleCCCResult(res *pipeline.Result) error {
+	if res != nil && res.OverflowingTx != nil {
+		if res.FinalBlock == nil {
+			// first txn overflowed the circuit, skip
+			log.Trace("Circuit capacity limit reached for a single tx", "tx", res.OverflowingTx.Hash().String(),
+				"isL1Message", res.OverflowingTx.IsL1MessageTx(), "reason", res.CCCErr.Error())
+
+			// Store skipped transaction in local db
+			overflowingTrace := res.OverflowingTrace
+			if !w.config.StoreSkippedTxTraces {
+				overflowingTrace = nil
+			}
+			rawdb.WriteSkippedTransaction(w.eth.ChainDb(), res.OverflowingTx, overflowingTrace, res.CCCErr.Error(),
+				w.currentPipeline.Header.Number.Uint64(), nil)
+
+			if overflowingL1MsgTx := res.OverflowingTx.AsL1MessageTx(); overflowingL1MsgTx != nil {
+				rawdb.WriteFirstQueueIndexNotInL2Block(w.eth.ChainDb(), w.currentPipeline.Header.ParentHash, overflowingL1MsgTx.QueueIndex+1)
+			} else {
+				w.eth.TxPool().RemoveTx(res.OverflowingTx.Hash(), true)
+			}
+		} else if !res.OverflowingTx.IsL1MessageTx() {
+			// prioritize overflowing L2 message as the first txn next block
+			// no need to prioritize L1 messages, they are fetched in order
+			// and processed first in every block anyways
+			w.prioritizedTx = &prioritizedTransaction{
+				blockNumber: w.currentPipeline.Header.Number.Uint64() + 1,
+				tx:          res.OverflowingTx,
+			}
+		}
+
+		switch {
+		case res.OverflowingTx.IsL1MessageTx() &&
+			errors.Is(res.CCCErr, circuitcapacitychecker.ErrBlockRowConsumptionOverflow):
+			l1TxRowConsumptionOverflowCounter.Inc(1)
+		case !res.OverflowingTx.IsL1MessageTx() &&
+			errors.Is(res.CCCErr, circuitcapacitychecker.ErrBlockRowConsumptionOverflow):
+			l2TxRowConsumptionOverflowCounter.Inc(1)
+		case res.OverflowingTx.IsL1MessageTx() &&
+			errors.Is(res.CCCErr, circuitcapacitychecker.ErrUnknown):
+			l1TxCccUnknownErrCounter.Inc(1)
+		case !res.OverflowingTx.IsL1MessageTx() &&
+			errors.Is(res.CCCErr, circuitcapacitychecker.ErrUnknown):
+			l2TxCccUnknownErrCounter.Inc(1)
+		}
+	}
+
+	if !w.isRunning() {
+		if res != nil && res.FinalBlock != nil {
+			w.updateSnapshot(res.FinalBlock)
+		}
+		w.currentPipeline = nil
+		return nil
+	}
+
+	if res == nil || res.FinalBlock == nil {
+		w.startNewPipeline(time.Now().Unix())
```

Comment:
If `startNewPipeline` calls `handleCCCResult`, and then here we call `startNewPipeline`, will this second execution succeed (as opposed to the first one)?

Reply:
This case is for when the first txn fails the CC check. [The lines](https://github.com/scroll-tech/go-ethereum/pull/735/files#diff-1825330f8f75827945cb0144a6ab3dc034a3081f48ee96b0a318c5de9950832eR773-R791) above skips that txn. So the second time we create a pipeline hopefully we won't be getting another txn that we need to skip and be able to create a new block.